include/miral/miral/display_configuration.h

```diff
@@ -81,35 +138,74 @@ class DisplayConfiguration
         std::unique_ptr<Self> self;
     };
 
+    /// Construct the display configuration.
+    ///
+    /// The configuration file will be loaded from the provided mir runner via
+    /// #miral::MirRunner::display_config_file.
+    ///
+    /// The configuration is loaded by #DisplayConfiguration::operator()(mir::Server&).
+    ///
+    /// \param mir_runner the mir runner
     explicit DisplayConfiguration(MirRunner const& mir_runner);
 
-    /// Provide the default 'display-layout' configuration option
+    /// Create a `--display-layout` configuration option.
+    ///
+    /// Note that this must be provided to #miral::MirRunner::run_with in order to
+    /// work.
+    ///
+    /// \returns a new configuration option
     auto layout_option() -> ConfigurationOption;
 
-    /// Select a layout from the configuration
+    /// Select a layout from the configuration.
+    ///
+    /// If the \p layout is not present in the configuration, the 'default'
+    /// layout is applied.
+    ///
+    /// \param layout the desired layout
     void select_layout(std::string const& layout);
 
-    /// List all layouts found in the config file
+    /// List the valid layouts from the configuration file.
+    ///
+    /// \returns the list of valid layouts
     auto list_layouts() -> std::vector<std::string>;
 
-    /// Enable a custom output attribute in the .display YAML
+    /// Enable a custom output attribute in the configuration file.
+    ///
+    /// The value of this attribute will be made available through
+    /// #miral::UserDisplayConfigurationOutput::custom_attribute at the
+    /// \p key. Note that the data is value for the key has to have a
+    /// type of `std::optional<std::string>`.
```

Comment:
"Note that the data is value for the key has to have a type of" - not sure what this means? "that is the value for the key"?